gcloud-java-bigquery/src/main/java/com/google/gcloud/bigquery/Field.java

```diff
@@ -0,0 +1,370 @@
+          .add("fields", fields)
+          .toString();
+    }
+
+    @Override
+    public int hashCode() {
+      return Objects.hash(value, fields);
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+      if (!(obj instanceof Type)) {
+        return false;
+      }
+      Type other = (Type) obj;
+      return Objects.equals(value, other.value)
+          && Objects.equals(fields, other.fields);
+    }
+  }
+
+  /**
+   * Mode for a BigQuery Table field. {@link Mode#NULLABLE} fields can be set to {@code null},
+   * {@link Mode#REQUIRED} fields must be provided. {@link Mode#REPEATED} fields can contain more
+   * than one value.
+   */
+  public enum Mode {
+    NULLABLE, REQUIRED, REPEATED
+  }
+
+  private final String name;
+  private final Type type;
+  private final Mode mode;
+  private final String description;
+
+  public static final class Builder {
+
+    private String name;
+    private Type type;
+    private Mode mode;
+    private String description;
+
+    private Builder() {}
+
+    /**
+     * Sets the field name. The name must contain only letters (a-z, A-Z), numbers (0-9), or
+     * underscores (_), and must start with a letter or underscore. The maximum length is 128
+     * characters.
+     */
+    public Builder name(String name) {
+      this.name = checkNotNull(name);
+      return this;
+    }
+
+    /**
+     * Sets the value of the field.
+     *
+     * @see <a href="https://cloud.google.com/bigquery/preparing-data-for-bigquery#datatypes">
+     *     Data Types</a>
+     */
+    public Builder type(Type type) {
+      this.type = checkNotNull(type);
+      return this;
+    }
+
+    /**
+     * Sets the mode of the field. By default {@link Mode#NULLABLE} is used.
+     */
+    public Builder mode(Mode mode) {
+      this.mode = mode;
+      return this;
+    }
+
+    /**
+     * Sets the field description. The maximum length is 16K characters.
+     */
+    public Builder description(String description) {
+      this.description = description;
+      return this;
+    }
+
+    /**
+     * Creates an {@code Field} object.
+     */
+    public Field build() {
+      return new Field(this);
+    }
+  }
+
+  private Field(Builder builder) {
+    this.name = checkNotNull(builder.name);
+    this.type = checkNotNull(builder.type);
+    this.mode = builder.mode;
+    this.description = builder.description;
+  }
+
+  /**
+   * Returns the field name.
+   */
+  public String name() {
+    return name;
+  }
+
+  /**
+   * Returns the field value.
+   *
+   * @see <a href="https://cloud.google.com/bigquery/preparing-data-for-bigquery#datatypes">
+   *     Data Types</a>
+   */
+  public Type type() {
+    return type;
+  }
+
+  /**
+   * Returns the field mode. By default {@link Mode#NULLABLE} is used.
+   */
+  public Mode mode() {
+    return mode;
+  }
+
+  /**
+   * Returns the field description.
+   */
+  public String description() {
+    return description;
+  }
+
+  /**
```

Comment:
Though could be considered as a "convenient" method my preference would be not to have it.
Feel free to leave as is if you think otherwise.